client/src/components/ratingChart.jsx

```diff
@@ -1,24 +1,135 @@
+      star4: null,
+      star3: null,
+      star2: null,
+      star1: null
     };
+    this.getStarRatings = this.getStarRatings.bind(this);
+  }
+
+  componentDidMount () {
+    this.getStarRatings(this.props.productId);
+  }
+
+  getStarRatings (id) {
+    let productId = id;
+    axios.get(`http://localhost:4006/Reviews/getReviewSummary/${productId}`)
+      .then((response) => {
+        let starFive = parseInt(response.data.fiveStar.slice(0, response.data.fiveStar.length - 1));
+        let starFour = parseInt(response.data.fourStar.slice(0, response.data.fourStar.length - 1));
+        let starThree = parseInt(response.data.threeStar.slice(0, response.data.threeStar.length - 1));
+        let starTwo = parseInt(response.data.twoStar.slice(0, response.data.twoStar.length - 1));
+        let starOne = parseInt(response.data.oneStar.slice(0, response.data.oneStar.length - 1));
+        this.setState({
+          star5: starFive,
+          star4: starFour,
+          star3: starThree,
+          star2: starTwo,
+          star1: starOne
+        })
+          .catch(err => {
+            console.log(err);
+          });
+      });
   }
 
-  render () {
+
+  render() {
+
+    const Chart = styled.div`
+      display: flex;
+      flex-direction: column;
+      border-bottom: 1px solid grey;
+    `;
+
+    const StarBox = styled.button`
+      width: 180px;
+      height: 20px;
+      margin: 5px;
+      line-height: 5px;
+      border-radius: 5px;
+    `;
+
+    const Bar = styled.div`
+    display: inline-block;
+  `;
+
+    const Star5 = styled(StarBox)`
+      background: linear-gradient(90deg, orange 0% ${this.state.star5}%, #D3D3D3  ${this.state.star5}% 100%);
+    `;
+
+    const Star4 = styled(StarBox)`
+      background: linear-gradient(90deg, orange 0% ${this.state.star4}%, #D3D3D3 ${this.state.star4}% 100%);
+    `;
+    const Star3 = styled(StarBox)`
+      background: linear-gradient(90deg, orange 0% ${this.state.star3}%, #D3D3D3 ${this.state.star3}% 100%);
+    `;
+
+    const Star2 = styled(StarBox)`
+      background: linear-gradient(90deg, orange 0% ${this.state.star2}%, #D3D3D3 ${this.state.star2}% 100%);
+    `;
+    const Star1 = styled(StarBox)`
+      background: linear-gradient(90deg, orange 0% ${this.state.star1}%, #DCDCDC ${this.state.star1}% 100%);
+      margin-bottom: 20px;
+    `;
+
+    const BlueTxt = styled.h6`
+      color: #007185;
+      font-family: Arial, sans-serif;
+      font-size: 14px;
+      line-height: 16px;
+        &:hover {
+        color: #C7511F;
+        cursor: pointer;
+        }
+       margin: 1px;
+       display: inline-block;
+    `;
+
     return (
       <div>
-        <HorizontalBar data={data}></HorizontalBar>
+        <Chart>
+          <Bar>
+            <BlueTxt>5 star</BlueTxt>
+            <Star5></Star5>
+            <BlueTxt>{this.state.star5}%</BlueTxt>
+          </Bar>
+          <Bar>
+            <BlueTxt>4 star</BlueTxt>
+            <Star4></Star4>
```

Comment:
This is for displaying the bar chart ?

Reply:
Yes, I had a very tough time creating a chart from using a react chart library. I decided to create one myself so it is a bit hardcoded, but it was the solution I came to for getting a chart to popup.